### PR TITLE
feat: add operations to mosaic

### PR DESCRIPTION
**Pull request by wittjosiah**
### Summary
🎨🚀🧹

This pull request adds support for different drag and drop operations and custom sorting of items in the `aurora-grid` package. It introduces a new hook called `useItemsWithPreview` that handles the preview and rearrangement of items in the mosaic container. It also updates the `Stack`, `Grid`, and `Kanban` views and their stories to use the new hook and the `MosaicOperation` type. It removes some unused types and props and simplifies some import statements. It enables the debug overlay for some stories to demonstrate the new features.

> _`Mosaic` changes_
> _Drag and drop with preview_
> _Autumn leaves sorting_

### Walkthrough
*  Add new types and re-export them from the `Mosaic` namespace to support different drag and drop operations and sorting of items within containers ([link](https://github.com/dxos/dxos/pull/4417/files?diff=unified&w=0#diff-a0e23bacd3ccca35f5d39c83197975c8ffee79617a575fe0c46e30857ff41611L7-R14),[link](https://github.com/dxos/dxos/pull/4417/files?diff=unified&w=0#diff-a0e23bacd3ccca35f5d39c83197975c8ffee79617a575fe0c46e30857ff41611R72-R77),[link](https://github.com/dxos/dxos/pull/4417/files?diff=unified&w=0#diff-4788a5361aafa9aeeacdf5855fff3169f4ca6aae20b4b20981fe528650fe0a01R17-R29),[link](https://github.com/dxos/dxos/pull/4417/files?diff=unified&w=0#diff-e88f8f7ff7854dedcd54d6c50fadd0ab5ecc58ac56aa3341711d716bb0ab0820R6),[link](https://github.com/dxos/dxos/pull/4417/files?diff=unified&w=0#diff-6946803befd05be24ba6f76d2ed0391c382c98f0cb00e506c885e07b2e80b15dL5-R13)).

### CODE REVIEW

File: packages/ui/aurora-grid/src/views/Tree/testing.tsx [hunk 1]
```diff
@@ -130,28 +131,26 @@ export const GraphTree = ({ id, graph = createGraph(), debug }: { id: string; gr
...
-  const handleOver = ({ active, over }: MosaicMoveEvent<number>) => {
-    return Path.parent(active.path) !== Path.parent(over.path);
-  };
-
   return (
     <Tree
       id={id}
       items={graph.root.children as TreeData[]}
-      onOver={handleOver}
+      // onOver={handleOver}
```
Comment (wittjosiah) [inline packages/ui/aurora-grid/src/views/Tree/testing.tsx:153]:
```suggestion
```

File: packages/ui/aurora-grid/src/mosaic/Container.tsx [hunk 2]
```diff
@@ -7,13 +7,26 @@ import React, { createContext, useEffect, CSSProperties, HTMLAttributes, PropsWi
...
 import { useMosaic } from './hooks';
-import { MosaicDataItem, MosaicDraggedItem, MosaicMoveEvent } from './types';
+import { MosaicDataItem, MosaicDraggedItem } from './types';
 
 export type MosaicTileOverlayProps = {
   grow?: boolean;
   debug?: boolean;
 };
 
+export type MosaicOperation = 'adopt' | 'copy' | 'rearrange' | 'refuse';
```
[the remark this answers is not in the document]
Comment (thure) [reply packages/ui/aurora-grid/src/mosaic/Container.tsx:17]: 💯

Comment (thure) [inline packages/ui/aurora-grid/src/mosaic/Container.tsx:17]:
```suggestion
export type MosaicOperation = 'adopt' | 'copy' | 'rearrange' | 'reject';
```
perhaps?

**Review by thure — APPROVED** [review]
Gorgeous ✨

**Pull request merged**